src/pybind/mgr/orchestrator/module.py

```diff
@@ -29,7 +30,41 @@ def nice_delta(now, t, suffix=''):
         return '-'
 
 
-def to_format(what, format: str, many: bool, cls):
+class Format(enum.Enum):
+    plain = 'plain'
+    json = 'json'
+    json_pretty = 'json-pretty'
+    yaml = 'yaml'
+
+
+class ServiceType(enum.Enum):
+    mon = 'mon'
+    mgr = 'mgr'
+    rbd_mirror = 'rbd-mirror'
+    crash = 'crash'
+    alertmanager = 'alertmanager'
+    grafana = 'grafana'
+    node_exporter = 'node-exporter'
+    prometheus = 'prometheus'
+    cephadm_exporter = 'cephadm-exporter'
+
```

Comment:
https://github.com/ceph/ceph/blob/a51f8f2612d5815a51d4ab242a9a90c153155686/src/python-common/ceph/deployment/service_spec.py#L382-L384

Reply:
i'd fix `python-common/ceph/deployment/service_spec.py` then, so it's typed. and can be fit into the annotation system. or do you suggest we should use `Literal` ?

Reply:
Your introduction of enums is awesome. I really like them. In fact my plan would be to move them into the orchestrator interface an make use of them in mgr/cephadm later.